app_integrations/apps/salesforce.py

```diff
@@ -0,0 +1,395 @@
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001bROAQ/LogFile",
+                    "LogDate": "2014-03-14T00:00:00.000+0000",
+                    "LogFileLength": 2692.0
+                },
+                {
+                    "attributes": {
+                        "type": "EventLogFile",
+                        "url": "/services/data/v32.0/sobjects/EventLogFile/0ATD000000001SdOAI"
+                    },
+                    "Id": "0ATD000000001SdOAI",
+                    "EventType": "Console",
+                    "LogFile": "/services/data/v32.0/sobjects/EventLogFile/0ATD00001SdOAI/LogFile",
+                    "LogDate": "2014-03-13T00:00:00.000+0000",
+                    "LogFileLength": 1345.0
+                }
+            ]
+        }
+        """
+        log_files = []
+        for event_type in self._SALESFORCE_EVENT_TYPES:
+            event_type_format = 'AND+EventType+=+\'{}\''.format(event_type)
+            url = self._SALESFORCE_QUERY_URL.format(
+                instance_url=self._instance_url,
+                api_version=self._latest_api_version,
+                query=self._SALESFORCE_QUERY_FILTERS,
+                start_time=self._SALESFORCE_CREATE_AFTER.format(self._last_timestamp),
+                event_type=event_type_format
+            )
+            success, response = self._make_get_request(url, self._auth_headers)
+            if not success:
+                LOGGER.error('Failed to get a list of log files.')
+                return log_files
+
+            if response.get('records'):
+                log_files.extend([record['LogFile'] for record in response['records']
+                                  if record.get('LogFile')])
+
+        LOGGER.debug('Retrived %d log files', len(log_files))
+        return log_files
+
+    def _fetch_event_logs(self, log_file_path):
+        """Get event logs by sending GET request to each log file location.
+
+        Args:
+            log_file_path (str): log file location.
+
+        Returns:
+            list: a list of event logs or None.
+        """
+        url = '{}/{}'.format(self._instance_url, log_file_path)
+        data = None
+        try:
+            success, resp = self._make_get_request(url, self._auth_headers)
+            if not success:
+                LOGGER.error('Failed to get event logs')
+                return None
+
+            data = StringIO.StringIO(resp)
+            reader = csv.DictReader(data)
+        except (SalesforceAppError, IOError, csv.Error):
+            LOGGER.exception('Failed to get event logs')
+            if data:
+                data.close()
+            return None
+        return [log for log in reader]
+
+    def _gather_logs(self):
+        """Gather all log events. There are 32 event types.
+
+        Returns:
+            list: A list of dictionaries contains log events.
+        """
+        self._request_token()
```

Comment:
it looks like `_request_token` returns a bool - can we do a check here and return if this returns False (instead of doing the check in `_get_latest_api_version` for the `_instance_url`, etc)?